### PR TITLE
Improve feedback form

### PR DESCRIPTION
**Pull request by Hans5958**
- When the feedback server has problems, all inputs are disabled and a new alert is shown.
- When no addons list supplied, the checkbox is disabled, not hidden.
- Small refactors here and there, (e.q. fixing tech debt on the checkbox element).

### CODE REVIEW

**Comment by mxmou** [comment]
> When no addons list supplied, the checkbox is disabled, not hidden.

Why? If something is not available, it should be hidden unless the reason why it's disabled is obvious or somehow explained.

File: static/assets/js/feedback.js [hunk 1]
```diff
@@ -15,38 +19,59 @@ const setStatus = (statusText, status) => {
...
+        element.disabled = true
+    });
+    setStatus(i18n.statusOffline, "danger")
+}
+
+fetch("https://scratchaddons-feedback.glitch.me/", {
+    mode: 'no-cors'
+})
+    .then(response => {
+        if (response.status == 200) setOffline()
```
Comment (RedGuy12) [inline static/assets/js/feedback.js:73]: I think you forgot to negate this

File: static/assets/js/feedback.js [hunk 2]
```diff
@@ -15,38 +19,59 @@ const setStatus = (statusText, status) => {
     element.classList.add(`alert-${status}`)
 }
 
-document.querySelector("#feedback-form").onsubmit = async event => {
+form.onsubmit = async event => {
```
Comment (RedGuy12) [inline static/assets/js/feedback.js:22]: Please use an event listener.

**Comment by Hans5958** [comment]
> > When no addons list supplied, the checkbox is disabled, not hidden.
> 
> Why? If something is not available, it should be hidden unless the reason why it's disabled is obvious or somehow explained.

My consideration is to still expose that an option is possible, but you may tight since there are no mentions on it.

What if I add a tooltip that people can hover on it when it is disabled? Are you cool with that?

**Comment by mxmou** [comment]
> What if I add a tooltip that people can hover on it when it is disabled? Are you cool with that?

That would be fine.

**Comment by Hans5958** [comment]
Tooltips is ready.

I will wait until tommorow to merge this, in case there are other concerns.

**Comment by WorldLanguages** [comment]
@Hans5958 We could add something like `https://scratch.mit.edu/scratch-addons-extension/feedback` to the extension so that users can click on it (and we need to be careful so users don't lose the feedback they already wrote)

**Pull request merged**